client/core/core.go

```diff
@@ -5124,70 +5126,80 @@ func (c *Core) initializeDEXConnections(crypter encrypt.Crypter) {
-			continue
-		}
+		wg.Add(1)
+		go func(dc *dexConnection) {
+			defer wg.Done()
+			c.initializeDEXConnection(dc, crypter)
+		}(dc)
+	}
 
-		// Unlock the bond wallet if a target tier is set.
-		if bondAssetID, targetTier, maxBondedAmt := dc.bondOpts(); targetTier > 0 {
-			c.log.Debugf("Preparing %s wallet to maintain target tier of %d for %v, bonding limit %v",
-				unbip(bondAssetID), targetTier, dc.acct.host, maxBondedAmt)
-			wallet, exists := c.wallet(bondAssetID)
-			if !exists || !wallet.connected() { // connectWallets already run, just fail
-				subject, details := c.formatDetails(TopicBondWalletNotConnected, unbip(bondAssetID))
-				var w *WalletState
-				if exists {
-					w = wallet.state()
-				}
-				c.notify(newWalletConfigNote(TopicBondWalletNotConnected, subject, details, db.ErrorLevel, w))
-			} else if !wallet.unlocked() {
-				err = wallet.Unlock(crypter)
-				if err != nil {
-					subject, details := c.formatDetails(TopicWalletUnlockError, dc.acct.host, err)
-					c.notify(newFeePaymentNote(TopicWalletUnlockError, subject, details, db.ErrorLevel, dc.acct.host))
-				}
-			}
-		}
+	wg.Wait()
+}
 
-		if dc.acct.authed() { // should not be possible with newly idempotent login, but there's AccountImport...
-			continue // authDEX already done
-		}
+// initializeDEXConnection connects to the DEX server in the conns map and
+// authenticates the connection.
+func (c *Core) initializeDEXConnection(dc *dexConnection, crypter encrypt.Crypter) {
+	if dc.acct.isViewOnly() {
+		return // don't attempt authDEX for view-only conn
+	}
 
-		// Pending bonds will be handled by authDEX. Expired bonds will be
-		// refunded by rotateBonds.
+	// Unlock before checking auth and continuing, because if the user
+	// logged out and didn't shut down, the account is still authed, but
+	// locked, and needs unlocked.
+	err := dc.acct.unlock(crypter)
+	if err != nil {
+		subject, details := c.formatDetails(TopicAccountUnlockError, dc.acct.host, err)
+		c.notify(newFeePaymentNote(TopicAccountUnlockError, subject, details, db.ErrorLevel, dc.acct.host)) // newDEXAuthNote?
+		return
+	}
 
-		// If the connection is down, authDEX will fail on Send.
-		if dc.IsDown() {
-			c.log.Warnf("Connection to %v not available for authorization. "+
-				"It will automatically authorize when it connects.", dc.acct.host)
-			subject, details := c.formatDetails(TopicDEXDisconnected, dc.acct.host)
-			c.notify(newConnEventNote(TopicDEXDisconnected, subject, dc.acct.host, comms.Disconnected, details, db.ErrorLevel))
-			continue
-		}
+	if dc.acct.isDisabled() {
+		return // For disabled account, we only want dc.acct.unlock above to initialize the account ID.
+	}
 
-		wg.Add(1)
-		go func(dc *dexConnection) {
-			defer wg.Done()
-			err := c.authDEX(dc)
+	// Unlock the bond wallet if a target tier is set.
+	if bondAssetID, targetTier, maxBondedAmt := dc.bondOpts(); targetTier > 0 {
```

Comment:
Where do we unlock the wallets for active/pending/expired bonds that are not the currently configured bond asset?

Reply:
Done in https://github.com/decred/dcrdex/pull/2946/commits/bda10a3ba4eb18140bbc0bdb1d1ff348fa14e07e